Cannot build with `arc4random_buf` on Linux glibc

### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

```bash
cat >a.c <<EOF
#include <stdlib.h>

int main(void) {
	int a[100];
	arc4random_buf(a, 100);
	return 0;
}
EOF
zig cc -o a -std=gnu11 -D_GNU_SOURCE=1 a.c -target x86_64-linux-gnu.2.35
```

Output:

```
ld.lld: error: undefined symbol: arc4random_buf
>>> referenced by a.c:5
>>>               /home/shuaiz/.cache/zig/o/1143050e0f6c9bc272947fe861f0e341/a.o:(main)
```

### Expected Behavior

Either the function `arc4random_buf` not declared or correctly linked.

For example:

```bash
zig cc -o a -std=gnu11 -D_GNU_SOURCE=1 a.c -lc -target x86_64-linux-musl
```

This would report `arc4random_buf` not declared.

```
a.c:5:2: error: call to undeclared function 'arc4random_buf'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    5 |         arc4random_buf(a, 100);
      |         ^
1 error generated.
```

and

```bash
zig cc -o a -std=gnu11 -D_GNU_SOURCE=1 a.c -lc -target x86_64-macos
```

This would success with no error.

--- a/lib/libc/include/generic-glibc/stdlib.h
+++ b/lib/libc/include/generic-glibc/stdlib.h
@@ -653,6 +653,11 @@ extern int lcong48_r (unsigned short int __param[7],
 		      struct drand48_data *__buffer)
      __THROW __nonnull ((1, 2));
 
+/*
+ * arc4random* symbols introduced in glibc 2.36:
+ * https://sourceware.org/git/?p=glibc.git;a=blob;f=NEWS;h=8420a65cd06874ee09518366b8fba746a557212a;hb=6f4e0fcfa2d2b0915816a3a3a1d48b4763a7dee2
+ */
+#  if (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 36) || __GLIBC__ > 2
 /* Return a random integer between zero and 2**32-1 (inclusive).  */
 extern __uint32_t arc4random (void)
      __THROW __wur;
@@ -665,6 +670,7 @@ extern void arc4random_buf (void *__buf, size_t __size)
    limit (exclusive).  */
 extern __uint32_t arc4random_uniform (__uint32_t __upper_bound)
      __THROW __wur;
+#  endif /* glibc v2.36 and later */
 # endif	/* Use misc.  */
 #endif	/* Use misc or X/Open.  */
 

